contracts/bprotocol/info/LiquidatorInfo.sol

```diff
@@ -33,42 +36,67 @@ contract LiquidatorInfo {
+    struct AccountInfo {
+        AvatarInfo avatarInfo;
+        CushionInfo cushionInfo;
+        LiquidationInfo liquidationInfo;
+    }
 
-        uint numTokens = ctoken.length;
+    function isIn(address[] memory array, address elm) internal pure returns(bool) {
+        for(uint i = 0 ; i < array.length ; i++) {
+            if(elm == array[i]) return true;
+        }
 
+        return false;
+    }
+
+    function getAvatarInfo(
+        Registry registry,
+        BComptroller bComptroller,
+        address[] memory cTokens,
+        uint[] memory priceFeed,
+        address avatar
+    ) 
+        public
+        returns(AvatarInfo memory info) 
+    {
+        require(cTokens.length == priceFeed.length, "cTokens-priceFeed-missmatch");
+
+        uint numTokens = cTokens.length;
+        address user = registry.ownerOf(avatar);
+
+        info.user = user;
         info.debtTokens = new address[](numTokens);
-        info.debtAmounts = new address[](numTokens);
+        info.debtAmounts = new uint[](numTokens);
         info.collateralTokens = new address[](numTokens);
-        info.collateralAmounts = new address[](numTokens);
+        info.collateralAmounts = new uint[](numTokens);
         info.weightedCollateralAmounts = new uint[](numTokens);
 
-        for(uint i = 0 ; i < numTokens ; i++) {
-            if(registry.cEther() == ctoken[i]) info.debtTokens[i] = ETH;
-            else info.debtTokens[i] = CTokenInterface(info.debtTokens[i]).underlying();
+        IComptroller comptroller = bComptroller.comptroller();
+        address[] memory assetsIn = comptroller.getAssetsIn(avatar);
 
-            address btoken = bComptroller.c2b(ctoken[i]);
-            info.debtAmounts[i] = IBToken(btoken).borrowBalanceCurrent(user);
-
-            info.collateralTokens[i] = ctoken[i];
-            info.collateralAmounts[i] = CTokenInterface(btoken).exchangeRateCurrent() * CTokenInterface(btoken).balanceOf(user) / 1e18;
-            uint CR = bComptroller.comptroller().markets(ctoken[i]);
+        for(uint i = 0 ; i < numTokens ; i++) {
+            if(registry.cEther() == cTokens[i]) 
+                info.debtTokens[i] = ETH;
+            else 
+                info.debtTokens[i] = address(CTokenInterface(info.debtTokens[i]).underlying());
+
+            address bToken = bComptroller.c2b(cTokens[i]);
+            info.debtAmounts[i] = IBToken(bToken).borrowBalanceCurrent(user);
+
+            info.collateralTokens[i] = cTokens[i];
+            info.collateralAmounts[i] = CTokenInterface(bToken).exchangeRateCurrent() * CTokenInterface(bToken).balanceOf(user) / 1e18;
+            if(! isIn(assetsIn, cTokens[i])) info.collateralAmounts[i] = 0; 
+            // set as 0 if not in market
+            (,uint CR) = comptroller.markets(cTokens[i]);
             info.weightedCollateralAmounts[i] = info.collateralAmounts[i] * CR / 1e18;
 
             info.totalDebt += info.debtAmounts[i] * priceFeed[i] / 1e18;
```

Comment:
Should we have variable names like the following
- totalDebtValue / totalDebtInUSD
- totalCollateralValue / totalCollateralInUSD
- weightedCollateralValue / weightedCollateralInUSD